src/sage/rings/semirings/tropical_polynomial.py

```diff
@@ -0,0 +1,1016 @@
+        elif xmin is None or xmax is None:
+            raise ValueError("expected 2 inputs for xmin and xmax, but got 1")
+        elif xmin >= xmax:
+            raise ValueError(f"xmin = {xmin} should be less than xmax = {xmax}")
+        else:
+            return plot(f, xmin=xmin, xmax=xmax)
+
+    def _repr_(self):
+        r"""
+        Return a string represemtation of ``self``.
+
+        EXAMPLES::
+
+            sage: T = TropicalSemiring(QQ)
+            sage: R.<x> = PolynomialRing(T)
+            sage: R([-3,-1,2,-1])
+            (-1)*x^3 + 2*x^2 + (-1)*x + (-3)
+        """
+        import re
+        if not self.dict():
+            return str(self.parent().base().zero())
+
+        def replace_negatives(match):
+            return f'({match.group(0)})'
+
+        s = super()._repr()
+        v = self.parent().variable_name()
+        if s[0] == v:
+            s = "1*" + s
+        s = s.replace(" - ", " + -")
+        s = s.replace(" + "+v, " + 1*"+v)
+        s = s.replace("-"+v, "-1*"+v)
+        s = re.sub(r'-\d+', replace_negatives, s)
+        return s
+
+    def _latex_(self):
+        r"""
+        Return a latex representation of ``self``.
+
+        EXAMPLES::
+
+            sage: T = TropicalSemiring(QQ)
+            sage: R = PolynomialRing(T, 'x')
+            sage: p1 = R([-1,2,None,-3])
+            sage: latex(p1)
+            \left(-3\right) x^{3} + 2 x + \left(-1\right)
+        """
+        s = " "
+        coeffs = self.list(copy=False)
+        m = len(coeffs)
+        name = self.parent().latex_variable_names()[0]
+        for n in reversed(range(m)):
+            x = coeffs[n]
+            x = x._latex_()
+            if x != self.parent().base().zero()._latex_():
+                if n != m-1:
+                    s += " + "
+                if x.find("-") == 0:
+                    x = "\\left(" + x + "\\right)"
+                if n > 1:
+                    v = "|%s^{%s}" % (name, n)
+                elif n == 1:
+                    v = "|%s" % name
+                else:
+                    v = ""
+                s += "%s %s" % (x, v)
+        s = s.replace("|", "")
+        if s == " ":
+            return self.parent().base().zero()._latex_()
+        return s[1:].lstrip().rstrip()
+
+    def is_monic(self):
+        r"""
+        Return ``True`` if ``self`` is a monic tropical polynomial.
+
+        In the context of tropical polynomial, a monic polynomial has a
+        leading coefficient `0`.
+
+        EXAMPLES::
+
+            sage: T = TropicalSemiring(QQ)
+            sage: R.<x> = PolynomialRing(T)
+            sage: p1 = R([2,1,0]); p1
+            0*x^2 + 1*x + 2
+            sage: p1.is_monic()
+            True
+            sage: p2 = R([2,1,1]); p2
+            1*x^2 + 1*x + 2
+            sage: p2.is_monic()
+            False
+        """
+        if self[self.degree()] == self.base_ring().one():
+            return True
+        return False
```

Comment:
Why can't we just use the generic polynomial one after replacing the `self[self.degree()] == 1` with `== self.base().one()`?